File: src/main/java/com/flexpoker/table/command/handlers/TickActionOnTimerCommandHandler.java
package com.flexpoker.table.command.handlers;

import javax.inject.Inject;

import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Component;

import com.flexpoker.framework.command.CommandHandler;
import com.flexpoker.framework.pushnotifier.PushNotificationPublisher;
import com.flexpoker.pushnotifications.TickActionOnTimerPushNotification;
import com.flexpoker.table.command.commands.TickActionOnTimerCommand;

@Component
public class TickActionOnTimerCommandHandler implements CommandHandler<TickActionOnTimerCommand> {

    private final PushNotificationPublisher pushNotificationPublisher;

    @Inject
    public TickActionOnTimerCommandHandler(PushNotificationPublisher pushNotificationPublisher) {
        this.pushNotificationPublisher = pushNotificationPublisher;
    }

    @Async
    @Override
    public void handle(TickActionOnTimerCommand command) {
        var pushNotification = new TickActionOnTimerPushNotification(command.getGameId(),
                command.getTableId(), command.getNumber());
        pushNotificationPublisher.publish(pushNotification);
    }

}

File: src/main/kotlin/com/flexpoker/table/command/handlers/TickActionOnTimerCommandHandler.kt
package com.flexpoker.table.command.handlers

import com.flexpoker.framework.command.CommandHandler
import com.flexpoker.framework.pushnotifier.PushNotificationPublisher
import com.flexpoker.pushnotifications.TickActionOnTimerPushNotification
import com.flexpoker.table.command.commands.TickActionOnTimerCommand
import org.springframework.scheduling.annotation.Async
import org.springframework.stereotype.Component
import javax.inject.Inject

@Component
class TickActionOnTimerCommandHandler @Inject constructor(
    private val pushNotificationPublisher: PushNotificationPublisher
) : CommandHandler<TickActionOnTimerCommand> {

    @Async
    override fun handle(command: TickActionOnTimerCommand) {
        val pushNotification = TickActionOnTimerPushNotification(command.gameId, command.tableId, command.number)
        pushNotificationPublisher.publish(pushNotification)
    }

}